frontend/src/components/EmployeeCards/EmployeeCards.tsx

```diff
@@ -79,7 +57,7 @@ const EmployeeCard = ({
 
 
   return (
-    <Link to={{ pathname: '/employees/employee', state: userInfo, search: `?name=${fullName}` }}
+    <Link to={{ pathname: '/employees/employee/' + userInfo.id, state: userInfo }}
```

Comment:
you can make the pathname /admins or /drivers depending on the user, that way you wont have to make two requests later on in EmployeeDetail. just be sure to change the routes accordingly too